### PR TITLE
Linux

### PR DESCRIPTION
**Pull request by stoiver**
Updated pyswmm.lib to use linux shared library. Hopefully will not break the macos and windows code. Have added a linux build to travis CI.

### CODE REVIEW

**Comment by bemcdonnell** [comment]
@stoiver,  Thanks for the contributions!

@goanpeca, do we want to use Circle CI for linux testing instead of travis?

**Comment by goanpeca** [comment]
>do we want to use Circle CI for linux testing instead of travis?

Yes

**Comment by goanpeca** [comment]
And we are making a mixture of stuff... I don't think we should build SWMM here...

**Comment by stoiver** [comment]
I understand @goanpeca comment about not compiling swmm here. I was having trouble with getting a version of the linux library which didn't want to link in other libraries. But it seems to be working now.

File: .travis.yml [hunk 1]
```diff
@@ -1,17 +1,37 @@
...
+        brew update;
+        brew install gcc;
+      fi
+
+install: 
+  - pip install -r requirements-travis.txt
+
 script:
-  - pytest pyswmm
+  - py.test pyswmm --ignore=pyswmm/tests/test_plot_node_inflows.py --ignore=pyswmm/tests/test_plot_pump_setting.py --ignore=pyswmm/tests/test_plot_toolkit_units.py --ignore=pyswmm/tests/test_plot_weir_setting.py
```
Comment (bemcdonnell) [inline .travis.yml]: @stoiver, are these tests all failing?

Comment (bemcdonnell) [inline .travis.yml]: What was interesting to me, I was having segmentation faults for the set_nodeInflow function when I was not using malloc.  I wonder if that could actually fix the issue or not?

File: .travis.yml [hunk 2]
```diff
@@ -1,17 +1,37 @@
...
-    - "3.6"
-    - "2.7"
-sudo: false
 
 matrix:
   include:
     - os: osx
       language: generic
-# install dependecy
+      sudo: false
```
Comment (bemcdonnell) [inline .travis.yml]: I'm not sure if this is a huge deal or not but in Appveyor we are testing for py3.4 and py3.5 as well.

**Comment by bemcdonnell** [comment]
@stoiver, Awesome stuff here! Thanks!  

@goanpeca, regarding this issue (https://github.com/OpenWaterAnalytics/Stormwater-Management-Model/issues/4), how do you feel about setting up packaging before we get regression testing up and running?  We can make a note that currently the 32bit windows version is the canonical version.  

My concern at the moment is that as the toolkit API grows..  and if we continue to distribute the binaries with pyswmm, we are going to quickly get out of sync across the platforms.

**Comment by stoiver** [comment]
.travis.yml<https://github.com/OpenWaterAnalytics/pyswmm/pull/103#discussion_r120026492>:

>  script:
-  - pytest pyswmm
+  - py.test pyswmm --ignore=pyswmm/tests/test_plot_node_inflows.py --ignore=pyswmm/tests/test_plot_pump_setting.py --ignore=pyswmm/tests/test_plot_toolkit_units.py --ignore=pyswmm/tests/test_plot_weir_setting.py


@stoiver<https://github.com/stoiver>, are these tests all failing?


@bemcdonnell, These were failing on linux with a DISPLAY not set error, because of the plot matplotlib commands. Should be able to recode the tests to avoid this.

________________________________

In .travis.yml<https://github.com/OpenWaterAnalytics/pyswmm/pull/103#discussion_r120026556>:

>  script:
-  - pytest pyswmm
+  - py.test pyswmm --ignore=pyswmm/tests/test_plot_node_inflows.py --ignore=pyswmm/tests/test_plot_pump_setting.py --ignore=pyswmm/tests/test_plot_toolkit_units.py --ignore=pyswmm/tests/test_plot_weir_setting.py


What was interesting to me, I was having segmentation faults for the set_nodeInflow function when I was not using malloc. I wonder if that could actually fix the issue or not?


@bemcdonnell, The version of the swmm5.so I loaded is using the old code, and not the malloc version. On linux I was getting segmentation faults when USING malloc.



________________________________

In .travis.yml<https://github.com/OpenWaterAnalytics/pyswmm/pull/103#discussion_r120026648>:

>
 matrix:
   include:
     - os: osx
       language: generic
-# install dependecy
+      sudo: false


I'm not sure if this is a huge deal or not but in Appveyor we are testing for py3.4 and py3.5 as well.


@bemcdonnell, easy enough to add, I was getting enough errors to start with that I just stuck with 2.7 and 3.6 on osx and linux.

—

**Comment by bemcdonnell** [comment]
@stoiver, I just looked at those tests and they need to be updated to covert the high-level pyswmm functions.  Those were originally written to interact with the low-level pyswmm api (swmm5.py).  I will raise an issue and update those tests.  Feel free to leave it as is for now and I can update the travis file later. 

I think we're about ready to merge.  I'd like to see if @goanpeca has any thoughts on the PR

**Comment by goanpeca** [comment]
We can merge as is, and I can update the CI systems afterwards on PR #101

**Pull request merged**